helix-term/src/commands/lsp.rs

```diff
@@ -1100,6 +1100,56 @@ pub fn goto_reference(cx: &mut Context) {
     );
 }
 
+pub fn goto_reference_direction(cx: &mut Context, direction: Direction) {
+    let (view, doc) = current!(cx.editor);
+    let language_server = language_server!(cx.editor, doc);
+    let offset_encoding = language_server.offset_encoding();
+
+    let pos = doc.position(view.id, offset_encoding);
```

Comment:
Compared to all the other `]` commands, this is the only one that relies heavily on the cursor, I am not sure if this is a good idea, I think it would be good for user to first mark and select it with `<space>r` then only allow `]r`, otherwise user move cursor around and they have to get back to the same position to do `]r`.

Reply:
that would make this very similar to the already existing cmd implemented by #2738 . i don't think we would want to duplicate the same functionality across two different cmds, but rather let this be useful for jumping without having to select first if a user chooses to and use the existing `S-h` + `)/(` if they would rather select first, then jump